com.unity.render-pipelines.core/Runtime/Common/DynamicArray.cs

```diff
@@ -70,6 +81,129 @@ public int Add(in T value)
             return index;
         }
 
+        /// <summary>
+        /// Adds the elements of the specified collection to the end of the DynamicArray.
+        /// </summary>
+        /// <param name="array">The array whose elements should be added to the end of the DynamicArray. The array itself cannot be null, but it can contain elements that are null, if type T is a reference type.</param>
+        public void AddRange(DynamicArray<T> array)
+        {
+            Reserve(size + array.size);
+            for (int i = 0; i < array.size; ++i)
+                Add(array[i]);
```

Comment:
Quick performance note, given that we do the reserve before, there should be no need of doing the full Add that does range check,  unless I missed something.

Reply:
Good point. Updated the PR.